### PR TITLE
planning:modify the obstacles' interface for navi_path_decider and na…

### PR DESCRIPTION
**Pull request by konyhl**
@yifeijiang
 modify the obstacles' interface for navi_path_decider and navi_speed_decider.

### CODE REVIEW

File: modules/planning/common/planning_gflags.cc [hunk 1]
```diff
@@ -180,6 +180,8 @@ DEFINE_bool(enable_nudge_decision, true, "enable nudge decision");
 DEFINE_bool(enable_nudge_slowdown, true,
             "True to slow down when nudge obstacles.");
 
+DEFINE_bool(noenable_side_radar, true,
```
Comment (lianglia-apollo) [inline modules/planning/common/planning_gflags.cc]: Please use enable_side_rader

File: modules/planning/reference_line/reference_line_provider.cc [hunk 2]
```diff
@@ -480,10 +480,14 @@ bool ReferenceLineProvider::GetNearestWayPointFromNavigationPath(
     double s = 0.0;
     double l = 0.0;
     if (!lane->GetProjection({point.x(), point.y()}, &s, &l)) {
+      AERROR << "get adc projection point on lane failed";
       continue;
     }
     constexpr double kEpsilon = 1e-6;
-    if (s > (lane->total_length() + kEpsilon) || (s + kEpsilon) < 0.0) {
+    if (s > (lane->total_length() + kEpsilon)) {
```
Comment (lianglia-apollo) [inline modules/planning/reference_line/reference_line_provider.cc]: Why removed the second condition?

Comment (konyhl) [reply modules/planning/reference_line/reference_line_provider.cc]: Under the relative map, the initial s does not always start from 0.0, and many times the s of the first few points is negative.
if do't remove ,have maney such error:
reference_line_provider.cc:537] failed to find nearest point x: 0 y: 0

Comment (lianglia-apollo) [reply modules/planning/reference_line/reference_line_provider.cc]: Is it possible to treat this issue inside navigation mode? Other modules could be affected if there are changes on shared API.

Comment (konyhl) [reply modules/planning/reference_line/reference_line_provider.cc]: There is no suitable local revision at present,
i will revoke this amendment.

**Comment by konyhl** [comment]
@Liangliang Zhang
I have revised it according to your opinion.

**Pull request merged**